spec/services/create_planned_disbursement_spec.rb

```diff
@@ -13,6 +13,26 @@
         expect(result.object.parent_activity).to eq(activity)
       end
 
+      context "when financial quarter and year are provided" do
+        it "sets the period start and end dates" do
+          financial_quarter = "1"
+          financial_year = "2020"
+          result = described_class.new(activity: activity).call(attributes: {financial_quarter: financial_quarter, financial_year: financial_year})
+          expect(result.object.period_start_date).to eq "2020-04-01".to_date
+          expect(result.object.period_end_date).to eq "2020-06-30".to_date
+        end
+      end
+
+      context "when start and end date are provided" do
+        it "sets the financial quarter and year" do
+          start_date = "1 April 2020"
+          end_date = "30 June 2020"
+          result = described_class.new(activity: activity).call(attributes: {period_start_date: start_date, period_end_date: end_date})
+          expect(result.object.financial_quarter).to eq 1
+          expect(result.object.financial_year).to eq 2020
+        end
+      end
+
```

Comment:
What would the outcome be in the case of providing a start date that's on quarter 1, for example, and an end date that is on quarter 2?

Reply:
It's been a while since I looked at this code, but I suspect it would blow up! 😱 That said I am not sure you can provide dates that far apart through the validation rules - let me go check…

Reply:
Okay, took me a while to get my head back into this - sorry!

This spec is for the create_planned_disbursement service object which takes the start and end dates - the code that creates the financial quarter and financial year only uses the start date so would never go wrong.

The rule is 'if we have a start and end date, we'll take the start date and calculate the financial quarter and year from that ignoring the end date'.

Worth noting that both this test and the code are removed by the end of this PR.

Reply:
Okay. I think I went to the develop branch and tried to put start date and end date in different quarters and it allowed me. But as you said, in order to get the financial quarter we only need the start date. That solves the problem with the dates (plus this is outdated, so it doesn't matter much. I was just being curious, sorry!)

Reply:
Curious = good! 😄